### PR TITLE
Add fifo compaction options

### PR DESCRIPTION
**Pull request by arthurprs**
Add fifo compaction options.

### CODE REVIEW

**Comment by siddontang** [comment]
Thanks @arthurprs 

PTAL @zhangjinpeng1987 @huachaohuang

File: src/rocksdb_options.rs [hunk 1]
```diff
@@ -740,6 +740,12 @@ impl DBOptions {
         }
     }
 
+    pub fn set_fifo_compaction_options(&mut self, fifo_opts: &FifoCompactionOptions) {
```
Comment (siddontang) [inline src/rocksdb_options.rs]: why not moving FifoCompactionOptions here directly?

File: librocksdb_sys/src/lib.rs [hunk 2]
```diff
@@ -407,6 +408,10 @@ extern "C" {
         a2: i32,
     );
     pub fn crocksdb_options_set_compaction_style(options: *mut Options, cs: DBCompactionStyle);
+    pub fn crocksdb_options_set_fifo_compaction_options(
+        options: *mut Options,
+        cf_opts: *mut DBFifoCompactionOptions,
```
Comment (huachaohuang) [inline librocksdb_sys/src/lib.rs]: s/cf_opts/fifo_opts

File: librocksdb_sys/Cargo.toml [hunk 3]
```diff
@@ -17,4 +17,4 @@ portable = []
 sse = []
 
 [build-dependencies]
-gcc = "0.3"
+cc = "1.0"
```
Comment (siddontang) [inline librocksdb_sys/Cargo.toml]: seem we have already used 1.0 now.

File: src/rocksdb_options.rs (hunk 1 above)
Comment (arthurprs) [reply src/rocksdb_options.rs]: I changed it to a move. I think I tried to follow the pattern from BlockBasedOptions.

File: librocksdb_sys/src/lib.rs (hunk 2 above)
Comment (arthurprs) [reply librocksdb_sys/src/lib.rs]: Thanks.

**Comment by siddontang** [comment]
LGTM

File: src/rocksdb_options.rs [hunk 4]
```diff
@@ -740,6 +740,12 @@ impl DBOptions {
         }
     }
 
+    pub fn set_fifo_compaction_options(&mut self, fifo_opts: FifoCompactionOptions) {
```
Comment (zhangjinpeng1987) [inline src/rocksdb_options.rs]: I think this function should belong to `ColumnFamilyOptions`.

Comment (arthurprs) [reply src/rocksdb_options.rs]: I agree. I think it was an oversight by rocksdb team. I think we shouldn't deviate from the original api though.

Comment (zhangjinpeng1987) [reply src/rocksdb_options.rs]: In this repo, we distinguish `DBOptions` and `ColumnFamilyOptions` although they both wrap RocksDB's `Options`.

Comment (arthurprs) [reply src/rocksdb_options.rs]: Good point. I changed it now.

**Comment by siddontang** [comment]
PTAL @zhangjinpeng1987 @huachaohuang

**Comment by siddontang** [comment]
Ping @zhangjinpeng1987 @huachaohuang

**Pull request merged**